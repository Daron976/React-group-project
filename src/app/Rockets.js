import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/RocketApi';
import RocketItems from './rocket-items';

const Rockets = () => {
  const dispatch = useDispatch();

  const rockets = useSelector((state) => state.rocketReducer);

  useEffect(() => {
    if (rockets.length !== 4) {
      dispatch(fetchRockets());
    }
  });

  return (
    <section className="rockets-holder">
      {rockets.map((rocket) => (
        <RocketItems
          key={rocket.id}
          img={rocket.flickr_images[0]}
          name={rocket.rocket_name}
          description={rocket.description}
        />
      ))}
    </section>
  );
};

export default Rockets;
