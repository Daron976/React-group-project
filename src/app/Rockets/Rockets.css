.each {
  display: flex;
  width: 90%;
  margin-inline: 5%;
  margin-bottom: 20px;
  border-radius: 15px;
  margin-top: 20px;
}

.each:hover {
  box-shadow: 0 0 10px black;
}

.pictFrame {
  width: 20%;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: hidden;
}

.pict {
  min-width: 100%;
  min-height: 100%;
  border-radius: 15px;
}

.infor {
  width: 80%;
  padding-left: 50px;
}

.reservRock {
  background-color: aquamarine;
  border-radius: 3px;
  padding: 2px 5px;
  font-size: 0.8rem;
}

.rocketBtn:hover {
  background-color: white !important;
  color: black !important;
}

button {
  padding: 0.5rem 1.5rem;
  background: none;
  border-radius: 5px;
  transition: all;
  transition-duration: 0.5s;
}

@media screen and (max-width: 768px) {
  .each {
    flex-direction: column;
    align-items: center;
    box-shadow: 0 0 10px black;
  }

  .pictFrame {
    width: 100%;
    border-radius: 15px;
  }

  .infor {
    margin-bottom: 2rem;
    padding: 0;
  }

  .Rocketslist {
    margin-top: 5rem;
  }
}
