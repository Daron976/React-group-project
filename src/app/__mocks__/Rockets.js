import Button from 'react-bootstrap/Button';
import PropTypes from 'prop-types';

const Rocket = (props) => {
  const { rocket } = props;
  return (
    <div className="each">
      <div className="pictFrame">
        <img
          src={rocket.flickr_images}
          alt={rocket.rocket_name}
          className="pict"
        />
      </div>
      <div className="infor">
        <h2>{rocket.rocket_name}</h2>
        <p>
          {rocket.reserved && <span className="reservRock">Reserved</span>}
          {rocket.description}
        </p>
        <Button
          variant={rocket.reserved ? 'outline-primary' : 'primary'}
          type="button"
          className="rocketBtn"
        >
          {rocket.reserved ? 'Cancel reservation' : 'Reserve Rockets'}
        </Button>
      </div>
    </div>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.shape({
    rocket_id: PropTypes.string,
    rocket_name: PropTypes.string,
    description: PropTypes.string,
    reserved: PropTypes.bool,
    flickr_images: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
};

const Rockets = () => {
  const rocketsData = [
    {
      id: 1,
      rocket_name: 'Falcon 1',
      description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009.',
      reserved: true,
      flickr_images: [
        'https://imgur.com/DaCfMsj.jpg',
        'https://imgur.com/azYafd8.jpg',
      ],
    },
  ];
  return (
    <div className="Rocketslist">
      {rocketsData.map((rocketData) => (
        <Rocket key={rocketData.id} rocket={rocketData} id={rocketData.id} />
      ))}
    </div>
  );
};

export default Rockets;
