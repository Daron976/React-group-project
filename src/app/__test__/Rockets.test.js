import renderer from 'react-test-renderer';
import Rockets from '../__mocks__/Rockets';

describe('Rockets', () => {
  test('Rockets container items renders as expected', async () => {
    const snapRockets = renderer.create(
      <Rockets />,
    )
      .toJSON();
    expect(snapRockets).toMatchSnapshot();
  });
});
