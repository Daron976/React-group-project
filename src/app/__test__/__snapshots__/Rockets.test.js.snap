// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Rockets container items renders as expected 1`] = `
<div
  className="Rocketslist"
>
  <div
    className="each"
  >
    <div
      className="pictFrame"
    >
      <img
        alt="Falcon 1"
        className="pict"
        src={
          Array [
            "https://imgur.com/DaCfMsj.jpg",
            "https://imgur.com/azYafd8.jpg",
          ]
        }
      />
    </div>
    <div
      className="infor"
    >
      <h2>
        Falcon 1
      </h2>
      <p>
        <span
          className="reservRock"
        >
          Reserved
        </span>
        The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009.
      </p>
      <button
        className="rocketBtn btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Cancel reservation
      </button>
    </div>
  </div>
</div>
`;
